digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/AwsService.scala

```diff
@@ -0,0 +1,299 @@
+                 val numberOfLanesService: NumberOfLanesService
+                ) {
+
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  def awsDao: AwsDao = new AwsDao
+
+  implicit val formats = Serialization.formats(NoTypeHints)
+
+  final val AwsUser = "AwsUpdater"
+
+  val allDatasetStatus = Map(
+    0 -> "Inserted successfuly",
+    1 -> "Amount of features and roadlinks do not match",
+    2 -> "Errors in the features",
+    3 -> "Processed successfuly",
+    4 -> "Errors while processing"
+  )
+
+  val allFeatureStatus = Map(
+    0 -> "Inserted successfuly",
+    1 -> "Errors while validating",
+    2 -> "Processed successfuly",
+    3 -> "SpeedLimit with invalid speed",
+    4 -> "PavementClass with invalid pavement class",
+    5 -> "Obstacle with invalid class",
+    6 -> "Invalid sideCode"
+  )
+
+  private def linkIdValidation(linkIds: Set[Long]): Unit = {
+    val roadLinks = roadLinkService.getRoadsLinksFromVVH(linkIds)
+    linkIds.foreach { linkId =>
+      roadLinks.find(road => road.linkId == linkId && road.administrativeClass != State).get
+    }
+  }
+
+  private def validatePoint(datasetId: String, featureId: Long, properties: Map[String, Any], assetType: String): Unit = {
+    assetType match {
+      case "obstacle" =>
+        try {
+          val obstacleType = properties.getOrElse("class", 0).asInstanceOf[BigInt]
+          if (!Set(1, 2).contains(obstacleType.intValue())) {
+            awsDao.updateFeatureStatus(featureId, "5")
+            awsDao.updateDatasetStatus(datasetId, 2)
+          }
+        } catch {
+          case _ =>
+            awsDao.updateFeatureStatus(featureId, "5")
+            awsDao.updateDatasetStatus(datasetId, 2)
+        }
+    }
+  }
+
+  private def validateRoadlink(datasetId: String, featureId: Long, properties: Map[String, Any]) = {
+    val speedLimit = properties.getOrElse("speedLimit", "").asInstanceOf[String]
+    val pavementClass = properties.getOrElse("pavementClass", "").asInstanceOf[String]
+    val sideCode = properties.getOrElse("sideCode", 0).asInstanceOf[BigInt]
+
+    speedLimit match {
+      case "" =>
+      case _ =>
+        if (!Set("20", "30", "40", "50", "60", "70", "80", "90", "100", "120").contains(speedLimit)) {
+          awsDao.updateFeatureStatus(featureId, "3")
+          awsDao.updateDatasetStatus(datasetId, 2)
+        }
+    }
+
+    pavementClass match {
+      case "" =>
+      case _ =>
+        if (!Seq("1", "2", "10", "20", "30", "40", "50").contains(pavementClass)) {
+          awsDao.updateFeatureStatus(featureId, "4")
+          awsDao.updateDatasetStatus(datasetId, 2)
+        }
+    }
+
+    sideCode match {
+      case _ =>
+        if (!(sideCode == 1)) {
+          awsDao.updateFeatureStatus(featureId, "6")
+          awsDao.updateDatasetStatus(datasetId, 2)
+        }
+    }
+  }
```

Comment:
It is better if we use get instead of getOrElse in the first 3 lines and then use match with options fetched in those 3 lines